gravitee-apim-console-webui/src/management/settings/portal-ng/portal-ng.component.ts

```diff
@@ -0,0 +1,473 @@
+      .subscribe((selectedValue) => {
+        if (!selectedValue) {
+          this.portalForm.get('portal.userCreation.automaticValidation.enabled').disable();
+        }
+        if (selectedValue) {
+          this.portalForm.get('portal.userCreation.automaticValidation.enabled').enable();
+        }
+      });
+
+    this.portalForm
+      .get('openAPIDocViewer.openAPIDocType.redoc.enabled')
+      .valueChanges.pipe(takeUntil(this.unsubscribe$))
+      .subscribe((selectedValue) => {
+        if (!selectedValue) {
+          this.portalForm.get('openAPIDocViewer.openAPIDocType.defaultType').setValue('Swagger');
+          this.portalForm.get('openAPIDocViewer.openAPIDocType.swagger.enabled').disable();
+        }
+        if (selectedValue) {
+          this.portalForm.get('openAPIDocViewer.openAPIDocType.swagger.enabled').enable();
+        }
+      });
+
+    this.portalForm
+      .get('openAPIDocViewer.openAPIDocType.swagger.enabled')
+      .valueChanges.pipe(takeUntil(this.unsubscribe$))
+      .subscribe((selectedValue) => {
+        if (!selectedValue) {
+          this.portalForm.get('openAPIDocViewer.openAPIDocType.defaultType').setValue('Redoc');
+          this.portalForm.get('openAPIDocViewer.openAPIDocType.redoc.enabled').disable();
+        }
+        if (selectedValue) {
+          this.portalForm.get('openAPIDocViewer.openAPIDocType.redoc.enabled').enable();
+        }
+      });
+  }
+
+  onSubmit() {
+    delete this.portalForm.value.portal.homepageTitleToggle;
+    const updatedSettingsPayload = {
+      ...this.settings,
+      company: {
+        ...this.settings.company,
+        ...this.portalForm.get('company').value,
+      },
+      plan: {
+        security: {
+          ...this.settings.plan.security,
+          ...this.portalForm.get('security').value,
+        },
+      },
+      api: {
+        ...this.settings.api,
+        ...this.portalForm.get('api').value,
+      },
+      dashboards: {
+        ...this.settings.dashboards,
+        ...this.portalForm.get('dashboards').value,
+      },
+      portal: {
+        ...this.settings.portal,
+        ...this.portalForm.get('portal').value,
+      },
+      scheduler: {
+        ...this.settings.scheduler,
+        ...this.portalForm.get('scheduler').value,
+      },
+      documentation: {
+        ...this.settings.documentation,
+        ...this.portalForm.get('documentation').value,
+      },
+      openAPIDocViewer: {
+        ...this.settings.openAPIDocViewer,
+        ...this.portalForm.get('openAPIDocViewer').value,
+      },
+      cors: {
+        ...this.settings.cors,
+        ...this.portalForm.get('cors').value,
+      },
+      email: {
+        ...this.settings.email,
+        ...this.portalForm.get('email').value,
+      },
+    };
+
+    this.portalSettingsService
+      .save(updatedSettingsPayload)
+      .pipe(
+        tap(() => this.snackBarService.success('Settings successfully updated!')),
+        takeUntil(this.unsubscribe$),
```

Comment:
What about replacing all of the `takeUntil(this.unsubscribe$)` with `takeUntilDestroyed()`?

Reply:
Done